exec/java-exec/src/main/java/org/apache/drill/exec/ops/QueryContext.java

```diff
@@ -97,7 +94,7 @@ public QueryContext(final UserSession session, final DrillbitContext drillbitCon
         plannerSettings.getPlanningMemoryLimit());
     bufferManager = new BufferManagerImpl(this.allocator);
     viewExpansionContext = new ViewExpansionContext(this);
-    schemaTreesToClose = Lists.newArrayList();
+    schemaTreeProvider = new SchemaTreeProvider(drillbitContext);
```

Comment:
Does this need to be created once per query? Can one instance live in Drillbit or UserSession?